pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1430,6 +1449,34 @@ CREATE OR REPLACE VIEW SCHEMA_INFO.label AS
       AS values
   FROM SCHEMA_CATALOG.label_key lk;
 
+CREATE OR REPLACE VIEW SCHEMA_INFO.system_stats AS
+    SELECT
+    (
+        SELECT SCHEMA_PROM.approximate_row_count('_prom_catalog.series'::REGCLASS)
+    ) AS num_series_approx,
+    (
+        SELECT count(*) FROM SCHEMA_CATALOG.metric
+    ) AS num_metric,
+    (
+        SELECT count(*) FROM SCHEMA_CATALOG.label_key
+    ) AS num_label_keys,
+    (
+        SELECT count(*) FROM SCHEMA_CATALOG.label
+    ) AS num_labels;
+
+CREATE OR REPLACE VIEW SCHEMA_INFO.metric_stats AS
+    SELECT metric_name,
+    SCHEMA_PROM.approximate_row_count(format('prom_series.%I', table_name)::regclass) AS num_series,
+    (SELECT row_estimate FROM public.hypertable_approximate_row_count(format('prom_data.%I',table_name)::regclass)) AS num_samples_approx
```

Comment:
For getting around the vanilla PG issue I suggest creating a wrapper function called SCHEMA_CATALOG.data_approximate_row_count() that calls either public.hypertable_approximate_row_count() or  SCHEMA_PROM.approximate_row_count() depending on SCHEMA_CATALOG.is_timescaledb_installed().

Search for `SCHEMA_CATALOG.is_timescaledb_installed()` for some examples.

Reply:
I over came this issue by making metric_stats to be created by a function where we can get more controls over pgsql commands. But this suggestion is definitely better. Thank you.

Reply:
how about naming `data_approximate_row_count` => `safe_approximate_row_count`?